community/consistency-check/src/main/java/org/neo4j/consistency/checking/SchemaRecordCheck.java

```diff
@@ -256,11 +258,22 @@ public void checkRelationshipPropertyExistenceRule( RelationshipPropertyExistenc
         }
     }
 
-    private void checkLabelAndPropertyRule( SchemaRule rule, int propertyKey, DynamicRecord record,
+    private void checkLabelAndPropertyRule( SchemaRule rule, EntityPropertyDescriptor descriptor, DynamicRecord record,
             RecordAccess records, CheckerEngine<DynamicRecord,ConsistencyReport.SchemaConsistencyReport> engine )
     {
         engine.comparativeCheck( records.label( rule.getLabel() ), VALID_LABEL );
-        engine.comparativeCheck( records.propertyKey( propertyKey ), VALID_PROPERTY_KEY );
+        if ( descriptor instanceof NodePropertyDescriptor && ((NodePropertyDescriptor) descriptor).isComposite() )
+        {
+            int[] propertyKeys = descriptor.getPropertyKeyIds();
```

Comment:
We actually did it that way first, and I commented above to Mattias about this. The current approach has the advantage of forcing runtime errors for when single and composite indexes get mixed up. It is just a way to enforce correct usage. However, I also see this a temporary or interim step while developing composite indexes to be sure single property indexes are tested as thoroughly as before. I think that as composite index features and tests for those features are created, we will simplify the descriptor framework further and aim to remove a lot of _if_ statements as a result.